ArcGISRuntimeSDKQt_CppSamples/EditData/DeleteFeaturesFeatureService/DeleteFeaturesFeatureService.cpp

```diff
@@ -109,35 +106,57 @@ void DeleteFeaturesFeatureService::connectSignals()
   });
 
   // connect to the identifyLayerCompleted signal on the map view
-  connect(m_mapView, &MapQuickView::identifyLayerCompleted, this, [this](QUuid, IdentifyLayerResult* identifyResult)
+  connect(m_mapView, &MapQuickView::identifyLayerCompleted, this, [this](QUuid, IdentifyLayerResult* rawIdentifyResult)
   {
+    // Deletes rawIdentifyResult instance when we leave scope.
+    QScopedPointer<IdentifyLayerResult> identifyResult(rawIdentifyResult);
+
     if(!identifyResult)
+    {
       return;
-    if (identifyResult->geoElements().size() > 0)
+    }
+
+    if (identifyResult->geoElements().isEmpty())
     {
-      // delete selected feature member if not nullptr
-      if (m_selectedFeature != nullptr)
-        delete m_selectedFeature;
+      return;
+    }
 
-      // select the item in the result
-      QueryParameters query;
-      query.setObjectIds(QList<qint64>() << identifyResult->geoElements().at(0)->attributes()->attributeValue("objectid").toLongLong());
-      m_featureLayer->selectFeatures(query, SelectionMode::New);
+    // delete selected feature member if not nullptr
+    if (m_selectedFeature != nullptr)
+    {
+      delete m_selectedFeature;
+      m_selectedFeature = nullptr;
+    }
 
-      // set selected feature member
-      m_selectedFeature = static_cast<ArcGISFeature*>(identifyResult->geoElements().at(0));
+    GeoElement* element = identifyResult->geoElements().at(0);
+    if (nullptr == element)
```

Comment:
`if (element == nullptr)` please. This is just style, but I don't think we compare nullptr on the left side anywhere else.